x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -611,7 +611,21 @@ class AgentPolicyService {
       );
     }
 
-    // Get updated agent policy
+    // Tamper protection is dependent on endpoint package policy
+    // Match tamper protection setting to the original policy
+    await this._update(
+      soClient,
+      esClient,
+      newAgentPolicy.id,
+      { is_protected: baseAgentPolicy.is_protected },
```

Comment:
couldn't this flag be added to the `create` call in line 565 to avoid doing an extra update?

Reply:
Despite initial talks of removing the `is_protected` field in the `create` api in the last agent tampering pr, we ended up deciding to keep it for simplicity and to prevent any potential breaking changes. So even though the `create` api accepts the `is_protected` field, unless the agent policy has the elastic defend package policy, we can't actually set `is_protected` in `create`.